### PR TITLE
The library stopped panicing on errors

### PR DESCRIPTION
**Pull request by artemkaxboy**

### CODE REVIEW

File: hocon/parser.go [hunk 1]
```diff
@@ -48,98 +59,138 @@ func (p *Parser) parseText(text string, callback IncludeCallback) *HoconRoot {
...
+
+			objectV, err := owner.GetObject()
+			if err != nil {
+				return err
+			}
+
+			if err := objectV.Merge(otherObj); err != nil {
+				return err
+			}
 		case TokenTypeEoF:
```
Comment (kubitre) [inline hocon/parser.go:129]: if u will have add to one case two different tokenType u should divided there by comma, like this: 
```golang
case TokenTypeEOF, TokenTypeKey:
```

File: hocon/tokenizer.go [hunk 2]
```diff
@@ -136,16 +142,20 @@ func (p *HoconTokenizer) PullRestOfLine() string {
...
-func (p *HoconTokenizer) PullNext() (token *Token) {
+func (p *HoconTokenizer) PullNext() (*Token, error) {
+	var token *Token
+	var err error
 
-	p.PullWhitespaceAndComments()
+	if err := p.PullWhitespaceAndComments(); err != nil {
+		return nil, err
+	}
 	if p.IsDot() {
```
Comment (kubitre) [inline hocon/tokenizer.go:159]: u need to think divided on how to get rid of such a long if-else ....

File: hocon/value.go [hunk 3]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
+		floatV, err := parsePositiveValue(foundFloat)
+		if err != nil {
+			return 0, err
+		}
 
-		switch u {
+		switch foundUnit {
 		case "nanoseconds", "nanosecond", "nanos", "nano", "ns":
-			return time.Duration(float64(time.Nanosecond) * v)
+			return time.Duration(float64(time.Nanosecond) * floatV), nil
```
Comment (kubitre) [inline hocon/value.go:568]: hm, it's very strange, because time.Duration - its type of int64 by any way how it work

File: hocon/parser.go [hunk 4]
```diff
@@ -48,98 +59,138 @@ func (p *Parser) parseText(text string, callback IncludeCallback) *HoconRoot {
...
+				return err
+			}
 			if !root {
-				return
+				return nil
 			}
 		case TokenTypeObjectEnd:
-			return
+			return nil
 		}
```
Comment (kubitre) [inline hocon/parser.go:144]: and where default case? if someone wants to break it, he just needs to change the value of TokenType to non exist token. by the way default case needed everywhere

File: hocon/parser.go [hunk 5]
```diff
@@ -48,98 +59,138 @@ func (p *Parser) parseText(text string, callback IncludeCallback) *HoconRoot {
...
 				}
 			}
-			p.ParseValue(value, true, currentPath)
-			return
+			return p.ParseValue(value, true, currentPath)
 		case TokenTypeObjectStart:
-			p.parseObject(value, true, currentPath)
-			return
+			return p.parseObject(value, true, currentPath)
 		}
```
Comment (kubitre) [inline hocon/parser.go:175]: and where default case? if someone wants to break it, he just needs to change the value of TokenType to non exist token. by the way default case needed everywhere

File: configuration.go [hunk 6]
```diff
@@ -4,42 +4,46 @@ import (
 	"encoding/json"
 	"io/ioutil"
 
-	"github.com/go-akka/configuration/hocon"
+	"github.com/goreflect/go_hocon/hocon"
 )
 
-func ParseString(text string, includeCallback ...hocon.IncludeCallback) *Config {
+func ParseString(text string, includeCallback ...hocon.IncludeCallback) (*Config, error) {
```
Comment (kubitre) [inline configuration.go:10]: i really don't imagine why need this CallBack way =) all cases which resolving by this library use by default callback, maybe we should exclude this callback?

Comment (kubitre) [reply configuration.go:10]: it's simply this code

File: config.go [hunk 7]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+			return nil, nil
 		}
 	}
-	return currentNode
+	return currentNode, nil
 }
 
-func (p *Config) GetBoolean(path string, defaultVal ...bool) bool {
-	obj := p.GetNode(path)
+func (p *Config) GetBoolean(path string, defaultVal ...bool) (bool, error) {
```
Comment (kubitre) [inline config.go:95]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 8]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 	if obj == nil {
-		return big.NewInt(-1)
+		return big.NewInt(-1), nil
 	}
 	return obj.GetByteSize()
 }
 
-func (p *Config) GetInt32(path string, defaultVal ...int32) int32 {
-	obj := p.GetNode(path)
+func (p *Config) GetInt32(path string, defaultVal ...int32) (int32, error) {
```
Comment (kubitre) [inline config.go:122]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 9]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 		}
-		return 0
+		return 0, nil
 	}
 	return obj.GetInt32()
 }
 
-func (p *Config) GetInt64(path string, defaultVal ...int64) int64 {
-	obj := p.GetNode(path)
+func (p *Config) GetInt64(path string, defaultVal ...int64) (int64, error) {
```
Comment (kubitre) [inline config.go:137]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 10]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 		}
-		return 0
+		return 0, nil
 	}
 	return obj.GetInt64()
 }
 
-func (p *Config) GetString(path string, defaultVal ...string) string {
-	obj := p.GetNode(path)
+func (p *Config) GetString(path string, defaultVal ...string) (string, error) {
```
Comment (kubitre) [inline config.go:152]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 11]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 		}
-		return ""
+		return "", nil
 	}
 	return obj.GetString()
 }
 
-func (p *Config) GetFloat32(path string, defaultVal ...float32) float32 {
-	obj := p.GetNode(path)
+func (p *Config) GetFloat32(path string, defaultVal ...float32) (float32, error) {
```
Comment (kubitre) [inline config.go:167]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 12]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+			return defaultVal[0], nil
 		}
+		return 0, nil
 	}
 	return obj.GetFloat32()
 }
 
-func (p *Config) GetFloat64(path string, defaultVal ...float64) float64 {
-	obj := p.GetNode(path)
+func (p *Config) GetFloat64(path string, defaultVal ...float64) (float64, error) {
```
Comment (kubitre) [inline config.go:182]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 13]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 		}
-		return 0
+		return 0, nil
 	}
 	return obj.GetFloat64()
 }
 
-func (p *Config) GetTimeDuration(path string, defaultVal ...time.Duration) time.Duration {
-	obj := p.GetNode(path)
+func (p *Config) GetTimeDuration(path string, defaultVal ...time.Duration) (time.Duration, error) {
```
Comment (kubitre) [inline config.go:197]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 14]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 		}
-		return 0
+		return 0, nil
 	}
 	return obj.GetTimeDuration(true)
 }
 
-func (p *Config) GetTimeDurationInfiniteNotAllowed(path string, defaultVal ...time.Duration) time.Duration {
-	obj := p.GetNode(path)
+func (p *Config) GetTimeDurationInfiniteNotAllowed(path string, defaultVal ...time.Duration) (time.Duration, error) {
```
Comment (kubitre) [inline config.go:212]: why accept many default value args? below only 1 element is used anyway

File: config.go [hunk 15]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-func (p *Config) GetValue(path string) *hocon.HoconValue {
+func (p *Config) GetValue(path string) (*hocon.HoconValue, error) {
 	return p.GetNode(path)
 }
 
-func (p *Config) WithFallback(fallback *Config) *Config {
+func (p *Config) WithFallback(fallback *Config) (*Config, error) {
 	if fallback == p {
-		panic("Config can not have itself as fallback")
+		return nil, nil
```
Comment (kubitre) [inline config.go:353]: maybe u will add same error?

File: config.go [hunk 16]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-	obj := p.GetNode(path)
+func (p *Config) GetFloat32List(path string) ([]float32, error) {
+	obj, err := p.GetNode(path)
+	if err != nil {
+		return nil, err
+	}
+
 	if obj == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:246]: need add error with description

File: hocon/parser.go [hunk 17]
```diff
@@ -210,21 +279,26 @@ func (p *Parser) ignoreNewline() {
...
-		currentNode = currentNode.GetChildObject(key)
+		var err error
+		currentNode, err = currentNode.GetChildObject(key)
+		if err != nil {
+			return nil, err
+		}
+
 		if currentNode == nil {
-			return nil
+			return nil, nil
```
Comment (kubitre) [inline hocon/parser.go:298]: without error description?

File: hocon/value.go [hunk 18]
```diff
@@ -103,130 +104,174 @@ func (p *HoconValue) topValueOfSub(v interface{}) HoconElement {
...
+	if err != nil {
+		return fmt.Sprintf("cannot get string: %s", err.Error())
+	}
+
+	return stringV
 }
 
-func (p *HoconValue) ToString(indent int) string {
+func (p *HoconValue) ToString(indent int) (string, error) {
 	if p.IsString() {
```
Comment (kubitre) [inline hocon/value.go:197]: this method need refactoring by switch case (change mechanism of definition about what object do u have)

File: hocon/value.go [hunk 19]
```diff
@@ -103,130 +104,174 @@ func (p *HoconValue) topValueOfSub(v interface{}) HoconElement {
...
 
-	return "<<unknown value>>"
+	return "<<unknown value>>", nil
 }
 
-func (p *HoconValue) GetObject() *HoconObject {
+func (p *HoconValue) GetObject() (*HoconObject, error) {
 	if len(p.values) == 0 {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline hocon/value.go:245]: add error description please

File: hocon/value.go [hunk 20]
```diff
@@ -103,130 +104,174 @@ func (p *HoconValue) topValueOfSub(v interface{}) HoconElement {
...
-		return o
+		return o, nil
 	}
 
 	raw = p.topValueOfSub(raw)
 
 	if s, ok := raw.(*HoconSubstitution); ok {
 		if s.ResolvedValue == nil {
-			return nil
+			return nil, nil
```
Comment (kubitre) [inline hocon/value.go:257]: description please

File: hocon/value.go [hunk 21]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
+
+	switch strings.ToLower(stringV) {
 	case "on", "true", "yes":
-		return true
+		return true, nil
 	case "off", "false", "no":
-		return false
-	default:
-		panic("Unknown boolean format: " + v)
+		return false, nil
```
Comment (kubitre) [inline hocon/value.go:302]: description

File: config.go [hunk 22]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+		if err != nil {
+			return nil, err
+		}
+
 		if currentNode == nil {
 			if p.fallback != nil {
 				return p.fallback.GetNode(path)
 			}
-			return nil
+			return nil, nil
```
Comment (kubitre) [inline config.go:89]: description

File: config.go [hunk 23]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+		return false, err
+	}
+
 	if obj == nil {
 		if len(defaultVal) > 0 {
-			return defaultVal[0]
+			return defaultVal[0], nil
 		}
-		return false
+		return false, nil
```
Comment (kubitre) [inline config.go:105]: description

File: config.go [hunk 24]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+		return 0, err
+	}
+
 	if obj == nil {
 		if len(defaultVal) > 0 {
-			return defaultVal[0]
+			return defaultVal[0], nil
 		}
-		return 0
+		return 0, nil
```
Comment (kubitre) [inline config.go:132]: description!

File: config.go [hunk 25]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
```
Comment (kubitre) [inline config.go:147]: description

File: config.go [hunk 26]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+		return "", err
+	}
+
 	if obj == nil {
 		if len(defaultVal) > 0 {
-			return defaultVal[0]
+			return defaultVal[0], nil
 		}
-		return ""
+		return "", nil
```
Comment (kubitre) [inline config.go:162]: description

File: config.go [hunk 27]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+	if err != nil {
+		return 0, err
+	}
+
 	if obj == nil {
 		if len(defaultVal) > 0 {
-			return defaultVal[0]
+			return defaultVal[0], nil
 		}
+		return 0, nil
```
Comment (kubitre) [inline config.go:177]: description

File: config.go [hunk 28]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+		return 0, err
+	}
+
 	if obj == nil {
 		if len(defaultVal) > 0 {
-			return defaultVal[0]
+			return defaultVal[0], nil
 		}
-		return 0
+		return 0, nil
```
Comment (kubitre) [inline config.go:192]: description

File: config.go [hunk 29]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
```
Comment (kubitre) [inline config.go:207]: description

File: config.go [hunk 30]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
```
Comment (kubitre) [inline config.go:222]: description

File: config.go [hunk 31]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-	obj := p.GetNode(path)
+func (p *Config) GetBooleanList(path string) ([]bool, error) {
+	obj, err := p.GetNode(path)
+	if err != nil {
+		return nil, err
+	}
+
 	if obj == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:234]: description

File: config.go [hunk 32]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-	obj := p.GetNode(path)
+func (p *Config) GetFloat64List(path string) ([]float64, error) {
+	obj, err := p.GetNode(path)
+	if err != nil {
+		return nil, err
+	}
+
 	if obj == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:258]: description

File: config.go [hunk 33]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-	obj := p.GetNode(path)
+func (p *Config) GetInt32List(path string) ([]int32, error) {
+	obj, err := p.GetNode(path)
+	if err != nil {
+		return nil, err
+	}
+
 	if obj == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:270]: description

File: config.go [hunk 34]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-	obj := p.GetNode(path)
+func (p *Config) GetInt64List(path string) ([]int64, error) {
+	obj, err := p.GetNode(path)
+	if err != nil {
+		return nil, err
+	}
+
 	if obj == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:282]: description

File: config.go [hunk 35]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-	obj := p.GetNode(path)
+func (p *Config) GetByteList(path string) ([]byte, error) {
+	obj, err := p.GetNode(path)
+	if err != nil {
+		return nil, err
+	}
+
 	if obj == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:294]: description

File: config.go [hunk 36]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
-	obj := p.GetNode(path)
+func (p *Config) GetStringList(path string) ([]string, error) {
+	obj, err := p.GetNode(path)
+	if err != nil {
+		return nil, err
+	}
+
 	if obj == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:306]: description

File: config.go [hunk 37]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 	}
 	return obj.GetStringList()
 }
 
-func (p *Config) GetConfig(path string) *Config {
+func (p *Config) GetConfig(path string) (*Config, error) {
 	if p == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:313]: description

File: config.go [hunk 38]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
 	if obj == nil {
-		return nil
+		return nil, nil
 	}
 	return obj.GetStringList()
 }
 
-func (p *Config) GetConfig(path string) *Config {
+func (p *Config) GetConfig(path string) (*Config, error) {
 	if p == nil {
```
Comment (kubitre) [inline config.go:312]: can this case really be done?

File: config.go [hunk 39]
```diff
@@ -62,218 +63,315 @@ func (p *Config) Copy(fallback ...*Config) *Config {
...
+			return nil, err
 		}
-		return NewConfigFromRoot(hocon.NewHoconRoot(value)).WithFallback(f)
+
+		return root.WithFallback(f)
 	}
 
 	if value == nil {
-		return nil
+		return nil, nil
```
Comment (kubitre) [inline config.go:342]: description

File: configuration.go [hunk 40]
```diff
@@ -4,42 +4,46 @@ import (
...
 
-func LoadConfig(filename string) *Config {
+func LoadConfig(filename string) (*Config, error) {
 	data, err := ioutil.ReadFile(filename)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 
 	return ParseString(string(data), defaultIncludeCallback)
```
Comment (kubitre) [inline configuration.go:31]: and why convert type of bytes to string? maybe we will change this way in another issue parsing by working only with bytes?

File: hocon/array.go [hunk 41]
```diff
@@ -16,22 +17,22 @@ func (p *HoconArray) IsString() bool {
 	return false
 }
 
-func (p *HoconArray) GetString() string {
-	panic("This element is an array and not a string.")
+func (p *HoconArray) GetString() (string, error) {
+	return "", errors.New("this element is an array and not a string")
 }
 
 func (p *HoconArray) IsArray() bool {
```
Comment (kubitre) [inline hocon/array.go:24]: I propose to get rid of this kind of methods and introduce constants for naming objects (then all places with conditions will be replaced by switches)

Comment (kubitre) [reply hocon/array.go:24]: really, see this: 
https://github.com/goreflect/go_hocon/pull/1/files?file-filters%5B%5D=.go#diff-5cdf89838be5e9a74f5ca743aa59e97fR21
Crazy man code which by use any method will be use bool methods for chosing case

File: hocon/parser.go [hunk 42]
```diff
@@ -167,35 +224,47 @@ func (p *Parser) ParseValue(owner *HoconValue, isEqualPlus bool, currentPath str
 		}
```
Comment (kubitre) [inline hocon/parser.go:225]: need add default case

File: hocon/value.go [hunk 43]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
+
+	return strconv.ParseInt(stringV, 10, 64)
 }
 
-func (p *HoconValue) GetInt32() int32 {
-	val, err := strconv.ParseInt(p.GetString(), 10, 32)
+func (p *HoconValue) GetInt32() (int32, error) {
+	stringV, err := p.GetString()
+	if err != nil {
+		return 0, nil
```
Comment (kubitre) [inline hocon/value.go]: maybe you wanted to return an error?

File: hocon/value.go [hunk 44]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
+
+	return int32(intV), nil
 }
 
-func (p *HoconValue) GetByte() byte {
-	val, err := strconv.ParseUint(p.GetString(), 10, 8)
+func (p *HoconValue) GetByte() (byte, error) {
+	stringV, err := p.GetString()
+	if err != nil {
+		return 0, nil
```
Comment (kubitre) [inline hocon/value.go]: ??

File: hocon/value.go [hunk 45]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
-		return nil
+func (p *HoconValue) GetChildObject(key string) (*HoconValue, error) {
+	objectV, err := p.GetObject()
+	if err != nil {
+		return nil, err
 	}
-	return obj.GetKey(key)
+
+	if objectV == nil {
+		return nil, nil
```
Comment (kubitre) [inline hocon/value.go:536]: description

File: hocon/value.go [hunk 46]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
+	}
+
+	return objectV.GetKey(key), nil
 }
 
 func (p *HoconValue) IsArray() bool {
```
Comment (kubitre) [inline hocon/value.go:542]: TRIGGERED

File: hocon/value.go [hunk 47]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
 		case "days", "day", "d":
-			return time.Duration(float64(time.Hour*24) * v)
+			return time.Duration(float64(time.Hour*24) * floatV), nil
 		}
+
+		return 0, fmt.Errorf("cannot parse time value: %s", stringV)
 	}
 
-	if strings.ToLower(res) == "infinite" {
+	if strings.ToLower(stringV) == "infinite" {
```
Comment (kubitre) [inline hocon/value.go]: "infinite" should be constant in special file with constants or at header of this file

File: hocon/value.go [hunk 48]
```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
...
 		if allowInfinite {
```
Comment (kubitre) [inline hocon/value.go:587]: it's very strange case

**Review by kubitre — COMMENTED** [review]
finished

File: hocon/parser.go (hunk 1 above)
Comment (artemkaxboy) [reply hocon/parser.go:129]: Done

File: configuration.go (hunk 6 above)
Comment (artemkaxboy) [reply configuration.go:10]: I believe we can simplify it in different issue. This PR is aimed to get rid of panics generally.

File: hocon/parser.go (hunk 1 above)
Comment (artemkaxboy) [reply hocon/parser.go:129]: Oops, sorry, it doesn't mean to be one of, it means do nothing in case of TokenTypeEoF

File: hocon/parser.go (hunk 17 above)
Comment (artemkaxboy) [reply hocon/parser.go:298]: It's not an error, it's just nil result

File: hocon/tokenizer.go (hunk 2 above)
Comment (artemkaxboy) [reply hocon/tokenizer.go:159]: I will)

File: hocon/value.go (hunk 19 above)
Comment (artemkaxboy) [reply hocon/value.go:245]: It's not an error

File: hocon/value.go (hunk 20 above)
Comment (artemkaxboy) [reply hocon/value.go:257]: Again, not an error

File: hocon/value.go (hunk 21 above)
Comment (artemkaxboy) [reply hocon/value.go:302]: Again not an error

File: hocon/value.go (hunk 43 above)
Comment (artemkaxboy) [reply hocon/value.go]: Yep, my fault. Fixed

File: hocon/value.go (hunk 44 above)
Comment (artemkaxboy) [reply hocon/value.go]: fixed

File: hocon/value.go (hunk 45 above)
Comment (artemkaxboy) [reply hocon/value.go:536]: not an error

File: hocon/value.go (hunk 3 above)
Comment (artemkaxboy) [reply hocon/value.go:568]: Hmm, not sure. Anyway, didn't touch it

File: hocon/value.go (hunk 47 above)
Comment (artemkaxboy) [reply hocon/value.go]: Done

**Comment by artemkaxboy** [comment]
@kubitre You made a few meaningful points about the logic and the structure of the program, but we should bare in mind, that this PR is called up to get rid of panics and we don't want to interfere in program logic unless we cover it all with tests.

File: hocon/parser.go (hunk 1 above)
Comment (kubitre) [reply hocon/parser.go:129]: Okay

**Pull request merged**